### PR TITLE
[IOBP-563, IOBP-612] Add empty states and navigation events to `PaymentsHomeScreen`

### PR DESCRIPTION
**Pull request by mastro993**
> [!WARNING]
> This PR depends on https://github.com/pagopa/io-app/pull/5659

## Short description
This PR introduces empty states and navigation events within the new payments landing screen

## List of changes proposed in this pull request
- Simplified `PaymentCardSmall` and `PaymentCardCarousel` components
- Added empty states to `PaymentHomeScreen`
- Added store reducer, actions and migration to manage empty methods banner visibility persistence
- Added navigation events to `PaymentHomeScreen` components
- Added unit tests
- Added missing locale keys
- Added payload to `paymentsDeleteMethodAction.success` in order to remove the wallet from the list once deleted successfully 

## How to test
Static check should solve correctly.
With the `io-dev-api-server`, enable te **New wallet section** from the **Profile** screen. 
Try to simulate different states by mocking api-server responses (`GET /ecommerce/wallets` and `GET /wallet/v1/transactions`). Check that the screen is rendered like in the previews below.

## Preview

|                      | **w/o payment methods** | **w/ payment methods** |
|----------------------|--------------------------|-------------------------|
| **w/o transactions** |   |                          |
| **w/ transactions**  |                           |                          |

### CODE REVIEW

File: ts/components/ui/utils/generateFlatListItemLayout.ts
Comment (dmnplb) [inline ts/components/ui/utils/generateFlatListItemLayout.ts:1]: Why was this file deleted? It might be useful to optimize the performance of `FlatList`

Comment (mastro993) [reply ts/components/ui/utils/generateFlatListItemLayout.ts:1]: That util function was making a lot of noise: an util function which return a function which returns an object for just a multiplication is overkill. 
I was planning to simplify it but then i forgot to put it back in the flat list!
Fixed in https://github.com/pagopa/io-app/pull/5663/commits/f316dd7d59fd3f9782059d65471c221fc667f401

File: ts/features/payments/home/components/PaymentsHomeEmptyScreenContent.tsx [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        { flexGrow: 1, paddingHorizontal: IOVisualCostants.appMarginDefault }
+      ]}
+    >
+      {withPictogram && (
+        <View testID="PaymentsHomeEmptyScreenContentTestID-pictogram">
+          <Pictogram name="empty" />
+          <VSpacer size={16} />
+        </View>
+      )}
+      <H4 style={{ textAlign: "center" }}>
```
Comment (hevelius) [inline ts/features/payments/home/components/PaymentsHomeEmptyScreenContent.tsx:45]: To get same code style in the code base what do you think to avoid inline style?

File: ts/features/payments/home/components/PaymentsHomeEmptyScreenContent.tsx [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        {I18n.t("features.payments.transactions.empty.title")}
+      </H4>
+      <VSpacer size={8} />
+      <Body style={{ textAlign: "center" }}>
```
Comment (hevelius) [inline ts/features/payments/home/components/PaymentsHomeEmptyScreenContent.tsx:49]: as above

**Review by hevelius — CHANGES_REQUESTED** [review]

File: ts/features/payments/home/components/PaymentsHomeEmptyScreenContent.tsx (hunk 2 above)
Comment (mastro993) [reply ts/features/payments/home/components/PaymentsHomeEmptyScreenContent.tsx]: I've improved the styling here https://github.com/pagopa/io-app/pull/5663/commits/c807808af1749cd812ec72c18e0369ba0a3d0771

**Review by hevelius — APPROVED** [review]
LGTM. Tested on iOS simulator. All cases works as expected.
